### ISSUE TITLE
Typescript errors when running in host app

### ISSUE THREAD

**Issue by Gaurav0** [ember-lifeline/ember-lifeline#332]
Upgraded my application to ember-cli-typescript@2.0.0-rc.2 from @1.5.0 and ember-cli-babel@7.4.1 from @6.18.0 and got the following errors:

```
ember s
node_modules/ember-lifeline/mixins/dom.d.ts:55:99 - error TS2304: Cannot find name 'RunMethod'.

55     addEventListener(this: MaybeIsComponent, target: TargetOrString, eventName: string, callback: RunMethod<MaybeIsComponent, any>, options?: Object | undefined): void;
                                                                                                     ~~~~~~~~~

node_modules/ember-lifeline/mixins/dom.d.ts:62:102 - error TS2304: Cannot find name 'RunMethod'.

62     removeEventListener(this: MaybeIsComponent, target: TargetOrString, eventName: string, callback: RunMethod<MaybeIsComponent, any>, options?: Object | undefined): void;
                                                                                                        ~~~~~~~~~
```

**Comment by Gaurav0** [comment]
Minimal Reproduction: https://github.com/Gaurav0/repro-lifeline

**Comment by buschtoens** [comment]
`RunMethod` is being referenced here:

https://github.com/ember-lifeline/ember-lifeline/blob/5ab67051e26f7f378a4cf3b698ce169743293bdc/addon/mixins/dom.ts#L68-L74

https://github.com/ember-lifeline/ember-lifeline/blob/5ab67051e26f7f378a4cf3b698ce169743293bdc/addon/mixins/dom.ts#L112-L118

It is defined here:

https://github.com/ember-lifeline/ember-lifeline/blob/5ab67051e26f7f378a4cf3b698ce169743293bdc/types/custom-ember.d.ts#L5-L7

Looking at this I wonder, if this ever worked in the first place? The `types` directory _is_ part of the npm tarball:

```bash
curl $(npm view ember-lifeline dist.tarball) | tar -ztf - 'package/types/*'
```

However, an addon's `types` directory is not included by the host app's `tsc`. You can "hotfix" this by adding `"node_modules/ember-lifeline/types/*"` to your `tsconfig.json` `"include"` array:

```bash
cat tsconfig.json \
  | jq '.include += ["node_modules/ember-lifeline/types/*"]' > tsconfig.json.fixed \
  && rm tsconfig.json \
  && mv tsconfig.json.fixed tsconfig.json
```

The proper fix is moving all of `custom-ember.d.ts` into `addon` and `import`ing its types explicitly where used.

**Comment by buschtoens** [comment]
I just tried running your reproduction with `ember-cli-typescript@1`, to make sure that this is not a regression is `@2` and I am getting the same error:

```
node_modules/ember-lifeline/mixins/dom.d.ts(55,99): error TS2304: Cannot find name 'RunMethod'.
node_modules/ember-lifeline/mixins/dom.d.ts(62,102): error TS2304: Cannot find name 'RunMethod'.
```

**Comment by rwjblue** [comment]
Confirm. We cannot use types/* for exported things. Since the consuming app won’t find them. 

We need to move the types to an imported location in addon folder I think.

**Comment by rwjblue** [comment]
Fixed by #412
